### PR TITLE
Issues/543

### LINKED ISSUE #534: support multiple countries for events
I will shortly add some initial support for this. I would like to have a map-based entry for regions i.e province/state (looks like this code was commented out with the change to Vue.js and we need to bring this back in @mehrdadgit ), and when that list is updated, the list of countries should also be updated if a new region.

I think it's best to allow operators to update this rather than having it automatically updated based on incoming reports as I had talked about with @LucieGueuning, as:
* A report may be spurious.
* An operator may want to flag a region / country for monitoring in anticipation of a report coming in.

### LINKED ISSUE #543: How to edit meta-data - generic info on event page?
for example, status or share point link?

### PR DESCRIPTION
**Pull request by qclin**
fixes #543 

inline editing for general information

rendering Region as tags, to be added & remove 
- I want to propose a region**s** field in metadata for model binding which may resolve #534 more elegantly

### CODE REVIEW

File: public/events/index.html [hunk 1]
```diff
@@ -591,13 +591,11 @@ <h4 class="modal-title" id="dialogModalTitle"></h4>
                 <div class="panel panel-default" >
 
                   <div class="panel-heading" style="cursor:pointer;" data-toggle="collapse" data-target="#collapseNotif" >
-                    <button type="button" class="btn btn-default btn-xs pull-right" @click="toggleEdit()">
-                      <span v-show="!editing"> Edit </span>
-                      <span v-show="editing"> Cancel </span>
-                    </button>
-                    <button type="button" v-show="editing" class="btn btn-primary btn-xs pull-right" @click="submitEventMetadata()"> Save </button>
+                    <button v-if="!panelEditing.Notification" type="button" class="btn btn-info btn-xs pull-right" @click="editEvent('General')">Edit</button>
```
Comment (matthewberryman) [inline public/events/index.html]: @qclin @mehrdadgit tagging this for later reference in GH PR comment

**Comment by matthewberryman** [comment]
I'm just getting onto this now that the weekend is over, which unfortunately has led to massive complication in trying to merge this: because @mehrdadgit's work on https://github.com/MSFREACH/msf-reach/issues/524 involved touching some of the same code around in-line editing, plus I've done some work on notifications too, separate to editing of general metadata by @qclin there were a large amount of merge conflicts. I've tried to resolve most in https://github.com/MSFREACH/msf-reach/pull/547/commits/c9a016db2b9fa417065d3fd87325bd5c3d2df8c6 although:
1. There were two different ways of doing the in-line editing buttons. I've tried to align these as best I can in https://github.com/MSFREACH/msf-reach/pull/547/commits/c9a016db2b9fa417065d3fd87325bd5c3d2df8c6#r205988087 
2. There are a lot of vue method functions done quite differently in https://github.com/MSFREACH/msf-reach/blob/issues/543/public/resources/js/events.js plus some added functionality in both sections (some changes around types and map code from @qclin in HEAD, some notification changes from myself and @mehrdadgit in dev plus some editing function differences) and I have not merged these.  Conflict marker starts https://github.com/MSFREACH/msf-reach/blob/c9a016db2b9fa417065d3fd87325bd5c3d2df8c6/public/resources/js/events.js#L1496 and ends https://github.com/MSFREACH/msf-reach/blob/c9a016db2b9fa417065d3fd87325bd5c3d2df8c6/public/resources/js/events.js#L1902 

Can you both please work on resolving that section and reviewing & UA testing the sections I have merged.

**Comment by qclin** [comment]
@matthewberryman 
hey i see the merge conflict is only with editEvent function 
which can be resolve by removing [https://github.com/MSFREACH/msf-reach/blob/c9a016db2b9fa417065d3fd87325bd5c3d2df8c6/public/resources/js/events.js#L1497-L1501](https://github.com/MSFREACH/msf-reach/blob/c9a016db2b9fa417065d3fd87325bd5c3d2df8c6/public/resources/js/events.js#L1497-L1501 )

as I'm merging the Maps functionality from Edit.html, it would be best to factor out as a separate js file to be call by both events/index.html & events/edit.html
since the transition from modal to inline editing is creating some duplicate code.

**Comment by mehrdadgit** [comment]
Hi @matthewberryman @qclin 
The event page is not working when I checked the branch out: 
image
* In general and based on my experience: as resolving conflicts is very confusing and time-consuming( sometimes takes longer than redoing the work for the GH issue) I suggest the person doing on the GH issue, merge latest dev into the issue branch, resolve all the conflicts, and make sure everything is working, before submitting a PR.

**Comment by matthewberryman** [comment]
Thanks - @qclin I'll leave it to you to merge events.js; can you please also test and then let me know when this PR is ready to go.

**Comment by qclin** [comment]
Hey sorry, I will keep in mind to merge dev next time before PR

anyway conflict is resolve via https://github.com/MSFREACH/msf-reach/commit/f9a375deafd3184c277463d7a6d64360971afa63

**Pull request merged**